app/services/check_course_jobs.rb

```diff
@@ -30,21 +48,43 @@ def delete_unique_lock
   def expected_digest
     hash = {
       'class' => 'CourseDataUpdateWorker',
-      'queue' => queue_for(@course),
+      'queue' => @queue,
       'unique_args' => [@course_id]
     }.to_json
     digest = OpenSSL::Digest::MD5.hexdigest hash
     "uniquejobs:#{digest}"
   end
 
-  def find_job
+  def find_job # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
Break this up into smaller methods instead off disabling the complexity cop. That will also make the code more self-documenting, if you choose good method names.

Reply:
Done!